projects/storefrontlib/src/shared/components/modal/modal.directive.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Directive,
+  HostListener,
+  Injectable,
+  Input,
+  Optional,
+} from '@angular/core';
+import { Router, RouterLink, RouterLinkWithHref } from '@angular/router';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+export interface ModalDirectiveOptions {
+  /**
+   * Action to perform after click
+   */
+  type: 'dismiss' | 'close';
+
+  /**
+   * Reason for dismissing or closing the modal
+   */
+  reason?: string;
+}
+
+/**
+ * Service to control the behavior of the DismissModalDirective
+ */
+@Injectable({
+  providedIn: 'root',
+})
```

Comment:
Service provided in root to allow for extensibility.
For example to add exceptions, when customer doesn't won't to close a modal, even if the directive has been applied.
Note: customer can't eaisly change the behavior of our built-in directives.